pkg/apis/core/validation/shoot_test.go

```diff
@@ -419,6 +419,39 @@ var _ = Describe("Shoot Validation Tests", func() {
 					Expect(ValidateShootHAConfigUpdate(newShoot, shoot)).To(BeEmpty())
 				})
 			})
+
+			Context("Shoot is hibernated", func() {
+				It("Should not allow upgrading from non-HA to HA when Spec.Hibernation.Enabled is set to `true`", func() {
+					shoot.Spec.ControlPlane = &core.ControlPlane{}
+					ValueTrue := true
+					newShoot := prepareShootForUpdate(shoot)
+					newShoot.Spec.ControlPlane = &core.ControlPlane{HighAvailability: &core.HighAvailability{FailureTolerance: core.FailureTolerance{Type: core.FailureToleranceTypeZone}}}
+					newShoot.Spec.Hibernation = &core.Hibernation{Enabled: &ValueTrue}
```

Comment:
we can get rid of ValueTrue
```suggestion
					newShoot.Spec.Hibernation = &core.Hibernation{Enabled: pointer.Bool(true)}
```